packages/server-core/src/projects/project/project.class.ts

```diff
@@ -133,6 +143,188 @@ const getGitProjectData = (project) => {
+    where: {
+      userId: userId,
+      type: 'github'
+    }
+  })
+
+  let repoPath = await getAuthenticatedRepo(githubIdentityProvider.oauthToken, data.sourceURL)
+  if (!repoPath) repoPath = data.sourceURL //public repo
+
+  const gitCloner = useGit(projectLocalDirectory)
+  await gitCloner.clone(repoPath, projectDirectory)
+  const git = useGit(projectDirectory)
+  const branchName = `${config.server.releaseName}-deployment`
+  try {
+    const branchExists = await git.raw(['ls-remote', '--heads', repoPath, `${branchName}`])
+    if (data.commitSHA) await git.checkout(data.commitSHA)
+    if (branchExists.length === 0 || data.reset) {
+      try {
+        await git.deleteLocalBranch(branchName)
+      } catch (err) {
+        //
+      }
+      await git.checkoutLocalBranch(branchName)
+    } else await git.checkout(branchName)
+  } catch (err) {
+    logger.error(err)
+    throw err
+  }
+
+  await uploadLocalProjectToProvider(app, projectName)
+
+  const projectConfig = getProjectConfig(projectName) ?? {}
+
+  // when we have successfully re-installed the project, remove the database entry if it already exists
+  const existingProjectResult = await app.service('project').Model.findOne({
+    where: {
+      [Op.or]: [
+        Sequelize.where(Sequelize.fn('lower', Sequelize.col('name')), {
+          [Op.like]: '%' + projectName + '%'
+        })
+      ]
+    }
+  })
+  let repositoryPath = data.destinationURL || data.sourceURL
+  const publicSignedExec = PUBLIC_SIGNED_REGEX.exec(repositoryPath)
+  //In testing, intermittently the signed URL was being entered into the database, which made matching impossible.
+  //Stripping the signed portion out if it's about to be inserted.
+  if (publicSignedExec) repositoryPath = `https://github.com/${publicSignedExec[1]}/${publicSignedExec[2]}`
+  const { commitSHA, commitDate } = await getCommitSHADate(projectName)
+  const returned = !existingProjectResult
+    ? // Add to DB
+      await app.service('project')._create(
+        {
+          thumbnail: projectConfig.thumbnail,
+          name: projectName,
+          repositoryPath,
+          needsRebuild: data.needsRebuild ? data.needsRebuild : true,
+          sourceRepo: data.sourceURL,
+          sourceBranch: data.sourceBranch,
+          updateType: data.updateType,
+          updateSchedule: data.updateSchedule,
+          updateUserId: userId,
+          commitSHA,
+          commitDate
+        },
+        params || {}
+      )
+    : await app.service('project')._patch(existingProjectResult.id, {
+        commitSHA,
+        commitDate,
+        sourceRepo: data.sourceURL,
+        sourceBranch: data.sourceBranch,
+        updateType: data.updateType,
+        updateSchedule: data.updateSchedule,
+        updateUserId: userId
+      })
+
+  returned.needsRebuild = typeof data.needsRebuild === 'boolean' ? data.needsRebuild : true
+
+  if (!existingProjectResult) {
+    await app.service('project-permission').create({
```

Comment:
This can be `_create` method, inorder to avoid calling resolvers, etc

Reply:
I want to call the feathers handler rather than the root create handler, since our project-permission.create handler automatically detects whether to make the calling user an owner of the project if there are no permissions holders on that project, e.g. if the project is newly-uploaded.

Reply:
Ohh okay